pkg/replication/replication.go

```diff
@@ -98,7 +98,7 @@ func (repl *Replicator) Enact(store IntentStore, serviceChecker ServiceChecker,
 				}
 				status, err := serviceStatus.ForNode(node.Name)
 
-				healthy, current := repl.isHealthyAndCurrent(status, err)
+				healthy, current := repl.isHealthyAndCurrent(node.Name, status, err)
```

Comment:
Shouldn't the right thing be to add ensure that `node.Name` is properly assigned to the resulting status, even if there's an error? Otherwise we're altering the `isHealthyAndCurrent` function just to support a log line.